src/Plugin/Field/FieldFormatter/IslandoraImageFormatter.php

```diff
@@ -103,39 +116,119 @@ public static function create(ContainerInterface $container, array $configuratio
       $container->get('current_user'),
       $container->get('entity_type.manager')->getStorage('image_style'),
       $container->get('islandora.utils'),
-      $container->get('file_url_generator')
+      $container->get('file_url_generator'),
+      $container->get('islandora.media_source_service')
     );
   }
 
+  /**
+   * {@inheritdoc}
+   */
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function defaultSettings() {
+    return [
+      'image_alt_text' => 'local',
+    ] + parent::defaultSettings();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsForm(array $form, FormStateInterface $form_state) {
+    $element = parent::settingsForm($form, $form_state);
+    $alt_text_options = [
+      'local' => $this->t('Local'),
+      'original_file_fallback' => $this->t('Local, with fallback to Original File'),
+      'original_file' => $this->t('Original File'),
+    ];
+    $element['image_alt_text'] = [
+      '#title' => $this->t('Alt text source'),
+      '#type' => 'select',
+      '#default_value' => $this->getSetting('image_alt_text'),
+      '#empty_option' => $this->t('None'),
+      '#options' => $alt_text_options,
+    ];
+    return $element;
+  }
+
   /**
    * {@inheritdoc}
    */
   public function viewElements(FieldItemListInterface $items, $langcode) {
     $elements = parent::viewElements($items, $langcode);
 
     $image_link_setting = $this->getSetting('image_link');
-    // Check if the formatter involves a link.
-    if ($image_link_setting != 'content') {
+    $alt_text_setting = $this->getsetting('image_alt_text');
+    // Check if we can leave the image as-is:
+    if ($image_link_setting != 'content' && $alt_text_setting == 'local') {
       return $elements;
     }
-
     $entity = $items->getEntity();
     if ($entity->isNew() || $entity->getEntityTypeId() != 'media') {
       return $elements;
     }
 
-    $node = $this->utils->getParentNode($entity);
-
-    if ($node === NULL) {
-      return $elements;
+    if ($alt_text_setting == 'none') {
+      foreach ($elements as $element) {
+        $element['#item']->set('alt', '');
+      }
     }
 
-    $url = $node->toUrl();
+    if ($image_link_setting == 'content' || $alt_text_setting == 'original_file' || $alt_text_setting == 'original_file_fallback') {
+      $node = $this->utils->getParentNode($entity);
+      if ($node === NULL) {
+        return $elements;
+      }
 
-    foreach ($elements as &$element) {
-      $element['#url'] = $url;
-    }
+      if ($image_link_setting == 'content') {
+        // Set image link.
+        $url = $node->toUrl();
+        foreach ($elements as &$element) {
+          $element['#url'] = $url;
+        }
+      }
+
+      if ($alt_text_setting == 'original_file' || $alt_text_setting == 'original_file_fallback') {
+        do {
```

Comment:
This doesn't actually loop, unsure of what the intent here is? Looks like you may just be able to remove it?

Reply:
Being in a `do` "loop" makes `break` possible. Otherwise I'd be in my natural state of writing 20 nested `if` statements (which is so hard to read, and i hear is considered not-good). And while I could replace each `break` with a `return`, that would make it harder for anyone who wants to add other code to this method to do so.